components/org.wso2.carbon.identity.governance/src/test/java/org/wso2/carbon/identity/governance/listener/IdentityMgtEventListenerTest.java

```diff
@@ -0,0 +1,577 @@
+import org.testng.IObjectFactory;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.ObjectFactory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.event.Event;
+import org.wso2.carbon.identity.event.services.IdentityEventService;
+import org.wso2.carbon.identity.governance.internal.IdentityMgtServiceDataHolder;
+import org.wso2.carbon.user.api.Permission;
+import org.wso2.carbon.user.api.RealmConfiguration;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.user.core.tenant.TenantManager;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.doReturn;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+
+/**
+ * Test class for IdentityMgtEventListener.
+ */
+public class IdentityMgtEventListenerTest {
+
+    private int SAMPLE_TENANT_ID = 3456;
+    private String SAMPLE_TENANT_DOMAIN = "abc.com";
+
+    @Mock
+    UserStoreManager userStoreManager;
+
+    @Mock
+    RealmConfiguration realmConfiguration;
+
+    @Mock
+    TenantManager tenantManager;
+
+    @Mock
+    RealmService realmService;
+
+    @Mock
+    IdentityEventService identityEventService;
+
+    IdentityMgtServiceDataHolder instanceIdentityDataHolder;
+    IdentityMgtServiceDataHolder instance;
+
+    @InjectMocks
+    @Spy
+    IdentityMgtEventListener identityMgtEventListener;
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    @BeforeTest
+    public void setUp() throws Exception {
+        instance = IdentityMgtServiceDataHolder.getInstance();
+        identityEventService = mock(IdentityEventService.class);
+        instance.setIdentityEventService(identityEventService);
+        realmService = mock(RealmService.class);
+        instance.setRealmService(realmService);
+        identityMgtEventListener = new IdentityMgtEventListener();
+        instanceIdentityDataHolder = new IdentityMgtServiceDataHolder();
+    }
+
+    @Test
+    public void testGetExecutionOrderId() {
+        int orderId = identityMgtEventListener.getExecutionOrderId();
+        assertEquals(orderId, 95, "OrderId is not equal to IdentityCoreConstants.EVENT_LISTENER_ORDER_ID");
```

Comment:
"OrderId is not equal to IdentityCoreConstants.EVENT_LISTENER_ORDER_ID" --->
"OrderId is not equal to " + IdentityCoreConstants.EVENT_LISTENER_ORDER_ID

Reply:
done